mobile/src/home-view.js

```diff
@@ -45,56 +68,201 @@ class HomeView extends PureComponent {
+    adminRef = fbc.database.public.adminRef('sessions')
     userRef = fbc.database.public.userRef()
     usersRef = fbc.database.public.usersRef()
     backgroundUrlRef = fbc.database.public.adminRef('backgroundUrl')
   }
 
   componentDidUpdate() {
-    const { sessionId, sessions } = this.state
+    const { sessionId, sessions, isAdminView } = this.state
     // If there is only one session, pick that automatically.
     if (!sessionId && Object.keys(sessions).length === 1) {
       this.setState({ sessionId: Object.keys(sessions)[0] })
     }
   }
 
   componentDidMount() {
+    client.getPrimaryColor().then(primaryColor => this.setState({ primaryColor }))
     client.getCurrentUser().then(currentUser => {
       this.setState({ currentUser })
-      this.signin.then(() => {
-        backgroundUrlRef.on('value', data => this.setState({ backgroundUrl: data.val() }))
-        sessionsRef.on('value', data => this.setState({ sessions: data.val() || {} }))
-        userRef.on('value', data => this.setState({ me: data.val() }))
-        this.answersRef().on('value', data => this.setState({ answers: data.val() || {} }))
-        mapPushedDataToStateObjects(usersRef, this, 'users')
-      })
+      this.signin
+        .then(() => {
+          backgroundUrlRef.on('value', data => this.setState({ backgroundUrl: data.val() }))
+          sessionsRef.on('value', data => this.setState({ sessions: data.val() || {} }))
+          userRef.on('value', data => this.setState({ me: data.val() }))
+          this.answersRef().on('value', data => this.setState({ answers: data.val() || {} }))
+          mapPushedDataToStateObjects(usersRef, this, 'users')
+
+          getAdmin = () => {
+            this.props.fbc.database.private
+              .adminRef('adminUrl')
+              .on('value', data => this.setState({ adminUrl: data.val() || undefined }))
+            adminSessionsRef.on('value', data => this.setState({ adminSessions: data.val() || {} }))
+          }
+
+          this.props.fbc.database.private
+            .adminableUserRef('adminToken')
+            .once('value', async data => {
+              const longLivedToken = data.val()
+              if (longLivedToken) {
+                await firebase.auth().signOut()
+                client.longLivedToken = longLivedToken
+                await this.props.fbc.signinAdmin()
+                console.log('Re-logged in as admin')
+                this.setState({ isAdmin: true })
+                getAdmin()
+              } else {
+                this.setState({ isAdminView: false })
+              }
+              this.hideLogInScreen = setTimeout(() => {
+                this.setState({ isLoggedIn: true })
+              }, 500)
+            })
+        })
+        .catch(() => this.setState({ logInFailed: true }))
+    })
+  }
+
+  switchView = () => {
+    const currentState = this.state.isAdminView
+    this.setState({
+      isAdminView: !currentState,
```

Comment:
This seems easier to read as a simple toggle to just say `isAdminView: !this.state.isAdminView`